### DISCUSSION TITLE
How to use iOS 13 Modals

### DISCUSSION THREAD

**Discussion by AAAstorga** [jonataslaw/getx#1052] [Q&A]
Hello,

How can I use  https://github.com/jamesblasco/modal_bottom_sheet? Is it possible? Thank you!

**Comment by fotiDim** [comment]
You just need to pass the context that you can easily get from GetX:
```
showCupertinoModalBottomSheet(
          context: Get.context,
          builder: (context) => Container(),
        ),
```

Make sure that you change your `CupertinoApp` to `GetCupertinoApp` or your `MaterialApp` to `GetMaterialApp` otherwise context will be null.

**Reply by eddyuan** [reply]
The bottom sheet works fine, but the existing page wouldn't shrink though. As the plugin indicates, you will need a `MaterialWithModalsPageRoute` in order to support the previous page transition, but Getx uses `GetPageRoute`, is there a way around?

**Reply by Ben8557** [reply]
Did you find the answer?

**Reply by cryptpa** [reply]
have you found answer :) ?

**Reply by rahi-staffasia** [reply]
Did you solve it?

**Reply by ghost** [reply]
Could not solve either.

**Reply by loic-hamdi** [reply]
@fotiDim did you find a solution?

**Reply by fotiDim** [reply]
@zzterrozz I am using only `showCupertinoModalBottomSheet` and nothing else. Works fine for me 🤷‍♂️. What is the issue you are having exactly?

**Reply by loic-hamdi** [reply]
@fotiDim I would like also to use `showCupertinoModalBottomSheet `
Would you mind pasting a code example for it?

**Reply by fotiDim** [reply]
```dart
var context = Get.context;
if (context != null) {
    showCupertinoModalBottomSheet(
        context: context,
        builder: (context) => const Container(),
    );
}
```
This is triggered within a `GetView` which I am using as a `drawer` in a `Scaffold`. My app is a `GetMaterialApp`. This is all I am doing. I can dismiss it by tap or swipe and it animates just fine.

**Reply by loic-hamdi** [reply]
I will try it out, Thank you @fotiDim 👍

**Reply by Albertbol** [reply]
@fotiDim problem is not that the modal bottom sheet not working, it does, but as it was written above problem is in the animation of the existing page:

> The bottom sheet works fine, but the existing page wouldn't shrink though.

**Comment by ghost** [comment]
+1
